### PR TITLE
[sdks] Change how we run test suites on the android app

### PR DESCRIPTION
**Pull request by luhenry**
Instead of using a textbox in the app, we use a file which contains which the name of the dll that contains the tests

### CODE REVIEW

**Comment by kumpera** [comment]
There's a reason to use a textbox, it's to help with XTC automation. We can't automate creating files on the target AFAICT.

**Comment by luhenry** [comment]
@kumpera I wasn't planning to dismiss your review, but what I will do is bring back the Activity which will be the one implementing support for testing on XTC. The Instrumentation will be used for tests on our CI with emulator and/or device.

**Comment by kumpera** [comment]
@luhenry no problem, just explaining why on earth there's a button in the UI. :)
I did that way cuz it's how Forms does XTC integration and I felt I had no reason to try something else.

**Comment by luhenry** [comment]
Makes sense. Have you had a chance to actually run and test it there? Also, how do you fetch back the results?

File: mcs/Makefile [hunk 1]
```diff
@@ -6,9 +6,9 @@ SUBDIRS := build jay mcs class nunit24 ilasm tools tests errors docs packages
 
 basic_SUBDIRS := build jay class
 build_SUBDIRS := build class class/aot-compiler tools ilasm
-monodroid_SUBDIRS := build class
+monodroid_SUBDIRS := build class tools
```
Comment (kumpera) [inline mcs/Makefile]: This is a lot nicer than having the sdk build go to that dir and a  "make PROFILE=xxx"!
I'll apply this to wasm as well.

File: sdks/android/Makefile [hunk 2]
```diff
@@ -79,3 +63,146 @@ logcat:
...
+package: $(APK_DIR)/assets/mconfig
+package: $(APK_DIR)/assets/asm/main.exe
+package: $(APK_DIR)/assets/asm/Mono.Android.dll
+package: $(patsubst %,$(APK_DIR)/assets/asm/%,$(BCL_LIBS))
+package: $(patsubst %,$(APK_DIR)/assets/asm/%,$(NUNIT_LIBS))
+package: $(patsubst %,$(APK_DIR)/assets/asm/monodroid_%_test.dll,$(CHECKS))
+
+.PHONY: build-%
+build-%: package
+	printf "monodroid_$*_test.dll" > $(APK_DIR)/assets/asm/testassembly.txt
```
Comment (kumpera) [inline sdks/android/Makefile]: This allows only a single test assembly per app. Why is perfect for local testing. OTOH, which XTC testing, having a single app makes it easy to manage execution and results collection.

File: sdks/android/Makefile [hunk 3]
```diff
@@ -79,3 +63,146 @@ logcat:
...
+
+.PHONY: check-%
+check-%: build-%
+	cd $(APK_DIR) && $(ADB) uninstall $(PACKAGE) || true
+	cd $(APK_DIR) && $(ADB) install bin/AndroidActivity-debug.apk
+	cd $(APK_DIR) && $(ADB) shell am instrument -w $(PACKAGE)/$(RUNNER)
+
+## Mini tests
+
+MINI_TEST_SOURCES= \
```
Comment (kumpera) [inline sdks/android/Makefile]: We should schedule work to move mini-test-runner.cs and the build rules to mono/mini and have it being built just like the BCL tests - IE, same targets and file names.

File: sdks/builds/android.mk [hunk 4]
```diff
@@ -26,7 +26,9 @@ _android_$(1)_RANLIB=$$(TOP)/sdks/builds/toolchains/android-$(1)/bin/$(3)-ranlib
 _android_$(1)_STRIP=$$(TOP)/sdks/builds/toolchains/android-$(1)/bin/$(3)-strip
 
 _android_$(1)_AC_VARS= \
-	mono_cv_uscore=yes
+	mono_cv_uscore=yes \
+	ac_cv_func_sched_getaffinity=no \
```
Comment (kumpera) [inline sdks/builds/android.mk]: This should be later moved to configure itself.

File: sdks/android/Makefile (hunk 2 above)
Comment (luhenry) [reply sdks/android/Makefile]: I will bring the XTC capability back.

**Pull request merged**